### PR TITLE
add kafka sasl configuration

### PR DESCRIPTION
**Pull request by ming-relax**
### What problem does this PR solve? 

add kafka sasl configuration

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below) ?
Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### CODE REVIEW

File: drainer/sync/kafka.go [hunk 1]
```diff
@@ -96,6 +96,12 @@ func NewKafka(cfg *DBConfig, tableInfoGetter translator.TableInfoGetter) (*Kafka
 	config.Producer.Retry.Max = 10000
 	config.Producer.Retry.Backoff = 500 * time.Millisecond
 
+	// sasl configuration
+	config.Net.SASL.Enable = cfg.KafkaSASLEnable
+	config.Net.SASL.Handshake = cfg.KafkaSASLHandshake
```
Comment (suzaku) [inline drainer/sync/kafka.go]: According to the documentation:

>You should only set this to false if you're using
>			// a non-Kafka SASL proxy.

I think we'd better make `KafkaSASLHandshake` `true` by default as in `sarama`.

File: drainer/sync/util.go [hunk 2]
```diff
@@ -29,11 +29,15 @@ type DBConfig struct {
...
-	KafkaMaxMessages int    `toml:"kafka-max-messages" json:"kafka-max-messages"`
-	TopicName        string `toml:"topic-name" json:"topic-name"`
+	ZKAddrs            string `toml:"zookeeper-addrs" json:"zookeeper-addrs"`
+	KafkaAddrs         string `toml:"kafka-addrs" json:"kafka-addrs"`
+	KafkaVersion       string `toml:"kafka-version" json:"kafka-version"`
+	KafkaMaxMessages   int    `toml:"kafka-max-messages" json:"kafka-max-messages"`
+	KafkaSASLEnable    bool   `toml:"kafka-sasl-enable" json:"kafka-sasl-enable"`
+	KafkaSASLHandshake bool   `toml:"kafka-sasl-handshake" json:"kafka-sasl-handshake"`
+	KafkaSASLUser      string `toml:"kafka-sasl-user" json:"kafka-sasl-user"`
+	KafkaSASLPassword  string `toml:"kafka-sasl-password" json:"kafka-sasl-password"`
```
Comment (suzaku) [inline drainer/sync/util.go]: Could you please also add some documentation for these new options?

1. `cmd/drainer/drainer.toml` acts like an example configuration file
1. https://github.com/pingcap/docs-cn/blob/master/tools/binlog/tidb-binlog-kafka.md is the documentation

**Comment by GregoryIan** [comment]
ping @ming-relax , please address comments

**Comment by GregoryIan** [comment]
I'll close this pr firstly, it's reserved for too long without update, but please take it easy to reopen it when you need it @ming-relax

**Pull request closed without merging**